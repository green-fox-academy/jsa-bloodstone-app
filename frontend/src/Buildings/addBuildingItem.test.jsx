import React from 'react';
import renderer from 'react-test-renderer';
import { TouchableHighlight } from 'react-native';

import AddBuildingItem from './addBuildingItem';

import addFarmIcon from '../../assets/buildings/addFarm.png';

describe('<AddBuildingItem />', () => {
  afterAll(() => {
    jest.resetModules();
  });

  it('Should render AddBuildingItem and match the snapshot', () => {
    const mockOnPress = jest.fn(() => {});

    const tree = renderer.create(<AddBuildingItem
      icon={addFarmIcon}
      type="test type"
      onPress={mockOnPress}
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should call onPress when press AddBuildingItem', () => {
    const mockOnPress = jest.fn(() => { });

    const testRenderer = renderer.create(<AddBuildingItem
      icon={addFarmIcon}
      type="test type"
      onPress={mockOnPress}
    />);
    const testInstance = testRenderer.root;

    testInstance.findByType(TouchableHighlight).props.onPress();
    expect(mockOnPress).toHaveBeenCalled();
  });
});
