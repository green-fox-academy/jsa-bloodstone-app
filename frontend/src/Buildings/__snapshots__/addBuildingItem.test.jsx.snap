// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddBuildingItem /> Should render AddBuildingItem and match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 2,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Image
      source={1}
      style={
        Object {
          "height": 60,
          "width": 60,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "fontSize": 12,
        "fontWeight": "bold",
        "lineHeight": 24,
        "marginBottom": -5,
      }
    }
  >
    Add test type
  </Text>
</View>
`;
